src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -351,21 +352,17 @@ private static boolean containsSummaryTag(DetailNode javadoc) {
      * @param javadoc javadoc root node.
      * @return first inline tag node or null if no node is found.
      */
-    private static DetailNode getFirstInlineTag(DetailNode javadoc) {
-        DetailNode node = null;
-        final DetailNode[] children = javadoc.getChildren();
-        for (DetailNode child: children) {
-            // If present as a children of javadoc
-            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
-                node = child;
-            }
-            // If nested inside html tag
-            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
-                node = getInlineTagNodeWithinHtmlElement(child);
-            }
-
-            if (node != null) {
-                break;
+    private static Optional<DetailNode> getFirstInlineTag(DetailNode javadoc) {
+        Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
+                .filter(child -> child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG)
+                .findFirst();
+
+        if (!node.isPresent()) {
+            final Optional<DetailNode> htmlNode = Arrays.stream(javadoc.getChildren())
+                    .filter(child -> child.getType() == JavadocTokenTypes.HTML_ELEMENT)
+                    .findFirst();
+            if (htmlNode.isPresent()) {
+                node = Optional.ofNullable(getInlineTagNodeWithinHtmlElement(htmlNode.get()));
```

Comment:
It seems like there is a bit of duplicated logic here (from the `getInlineTagNodeWithinHtmlElement()` method) that is being obfuscated by the use of `Optional`. Couldn't you just do something like:

```java

        final Predicate<DetailNode> childPredicate = child ->
                child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
                        || child.getType() == JavadocTokenTypes.HTML_ELEMENT;

        final Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
                .filter(childPredicate)
                .findFirst();

        return node.map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);

```

?

Reply:
Looks great. Thanks a lot.

Reply:
This change was causing an error, so had to revert back to my previous implementation with a slight change.

Reply:
@nmancus1 Please resolve if no more improvements.

Reply:
Resolved.